### PR TITLE
Added postgres conformance test for binding

### PR DESCRIPTION
**Pull request by amulyavarote**
Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

# Description

_Please explain the changes you've made_
Added postgres conformance test for binding.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1984 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/config/bindings/postgres/bindings.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: postgres-binding
+spec:
+  type: bindings.postgres
+  version: v1
+  metadata:
+  - name: url # Required
+    value: host=localhost user=postgres password=example port=5432 connect_timeout=10 database=dapr_test
```
Comment (berndverst) [inline tests/config/bindings/postgres/bindings.yml:10]: This isn't the correct format

**Review by berndverst — CHANGES_REQUESTED** [review]
Please add the quotation marks around the connection string.

File: .github/workflows/conformance.yml [hunk 2]
```diff
@@ -47,6 +47,7 @@ jobs:
         - bindings.mqtt-vernemq
         - bindings.redis
         - bindings.rabbitmq
+        - bindings.postgres
```
Comment (ItalyPaleAle) [inline .github/workflows/conformance.yml:50]: Nit, would be great if this list were sorted!

Comment (berndverst) [reply .github/workflows/conformance.yml:50]: Yes. Doesn't have to be in this PR though.

File: .github/workflows/conformance.yml [hunk 3]
```diff
@@ -299,6 +299,11 @@ jobs:
         docker-compose -f ./.github/infrastructure/docker-compose-postgresql.yml -p postgresql up -d
       if: contains(matrix.component, 'postgresql')
 
+    - name: Start postgres
```
Comment (berndverst) [inline .github/workflows/conformance.yml:302]: No, please remove. You can just update the name above and make it `postgres` which will match `postgresql` also :)

**Review by berndverst — CHANGES_REQUESTED** [review]

File: .github/workflows/conformance.yml (hunk 3 above)
Comment (amulyavarote) [reply .github/workflows/conformance.yml:302]: Yes. I will remove it. I am just testing a few things.

**Review by berndverst — APPROVED** [review]

**Pull request merged**